### PR TITLE
[ci] Automate the world (linting)

### PR DESCRIPTION
**Pull request by TheOtterlord**

### CODE REVIEW

**Comment by louisescher** [comment]
LGTM! My days of having to commit "Lint 💀" are finally over 🥲

**Review by louisescher — APPROVED** [review]

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: 'Format Code'
+
+on:
+  push:
```
Comment (Adammatthiesen) [inline .github/workflows/format.yml:4]: can we make this takeover the pull-request biome action as well?

https://github.com/astrolicious/studiocms/blob/7092e69d0eb1e4a4477444532117af9eab9736ed/.github/workflows/biome-linting.yml#L3-L20

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    branches:
+      - main
+
```
Comment (Adammatthiesen) [inline .github/workflows/format.yml:7]: dose this action need any specific permissions?

see the following for example:
https://github.com/astrolicious/studiocms/blob/7092e69d0eb1e4a4477444532117af9eab9736ed/.github/workflows/changeset-release.yml#L11-L24

File: .github/workflows/format.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          ref: ${{ github.head_ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v3
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
```
Comment (Adammatthiesen) [inline .github/workflows/format.yml:25]:
```suggestion
          node-version-file: '.node-version'
```
to conform with our repo and other actions

File: .github/workflows/format.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+          cache: pnpm
+      - name: Install dependencies
+        run: pnpm i
```
Comment (Adammatthiesen) [inline .github/workflows/format.yml:28]:
```suggestion
        run: pnpm ci:install
```
Wouldn't our `ci` install command be better here?

**Review by Adammatthiesen — CHANGES_REQUESTED** [review]
For the most part looking good, and i'm excited for this!

File: .github/workflows/format.yml (hunk 2 above)
Comment (TheOtterlord) [reply .github/workflows/format.yml:7]: probably

File: .github/workflows/format.yml (hunk 1 above)
Comment (TheOtterlord) [reply .github/workflows/format.yml:4]: also probably

**Review by louisescher — APPROVED** [review]
commit the fucker weeeeeeeeeeeeeeeeeeeeeeeeee

**Review by Adammatthiesen — APPROVED** [review]

**Pull request merged**